runtime/alloc.c

```diff
@@ -272,3 +272,11 @@ CAMLprim value caml_update_dummy(value dummy, value newval)
   }
   return Val_unit;
 }
+
+CAMLexport value caml_alloc_some(value v)
+{
+  CAMLparam1(v);
+  value some = caml_alloc_small(1, 0);
+  Store_field(some, 0, v);
```

Comment:
It'd be better to use direct assignment here:
```ocaml
Field(some, 0) = v;
```
since `Store_field` slows things down without any benefit in this context (immediately after calling `caml_alloc_small`).

Reply:
Please review #9819 :)

Reply:
Technically, doesn't GC rule 5 permit `caml_modify` to blow up with an uninitialized value (I know it doesn't at the moment)?

Reply:
It really does, and the old code can segfault! Well spotted, this was a bug.

Reply:
I don't think this can segfault, because the result of `caml_alloc_small` is in the minor heap, so the first `Is_young` test in `caml_modify` is true and a direct assignment is performed.  So, there is no need for an emergency backport to 4.11.  Still, this is wrong code according to the "GC rules" spelled out in the manual, and it should not have passed review.

Reply:
Ah, you're right.

Reply:
These changes aren't in 4.11, though, so there's no emergency either way

Reply:
> These changes aren't in 4.11, though

Excellent!  At some point I was really afraid the 4.11 release would have to be delayed.